client/src/js/game-channel/game-channel.middleware.js

```diff
@@ -1,21 +1,81 @@
 import * as DOMUpdate from '../dom-manipulation/dom-manipulation';
 
+/**
+ * @typedef {import("../game-channel/game-channel").GameChannel} gameChannel
+ */
+
+let resetPromise = null;
+
+/**
+ * @param {gameChannel} gameChannel
+ */
+const handleReset = async (gameChannel) => {
+  if (resetPromise) return;
+  resetPromise = setTimeout(() => {
+    gameChannel.shouldHandleReconnection = false;
+    DOMUpdate.setMoveStatus('bot', 'Waiting for user...');
+    DOMUpdate.resetSelections();
+    resetPromise = null;
+  }, 1000);
+};
+
+/**
+ * @param {gameChannel} gameChannel
+ */
 export function DomMiddleware(gameChannel) {
   if (gameChannel.isOpen) {
-    DOMUpdate.setParticipantBalance('user', gameChannel.balances.user);
-    DOMUpdate.setParticipantBalance('bot', gameChannel.balances.bot);
-  }
+    if (gameChannel.balances.user && gameChannel.balances.bot) {
+      DOMUpdate.setParticipantBalance('user', gameChannel.balances.user);
+      DOMUpdate.setParticipantBalance('bot', gameChannel.balances.bot);
+    }
+    DOMUpdate.setGameRoundIndex(gameChannel.gameRound.index);
+    DOMUpdate.setStakeAmount(gameChannel.gameRound.stake);
+    DOMUpdate.showGameInfo(false);
```

Comment:
```suggestion
    DOMUpdate.showGameInfo();
```

`showGameInfo` doesn't seem to take any args